in_toto/gpg.py

```diff
@@ -0,0 +1,247 @@
+    # on the header packet (that we will read right away)
+    other_headers_ptr = 0
+    ptr = 3
+    if packet_length == 1:
+        signature_length= struct.unpack(">H", data[1:ptr])[0]
+    else:
+        signature_length = data[1]
+        ptr = 2
+
+    # we get the version number, which we also expect to be v4, or we bail
+    # FIXME: support v4 type signatures (which I havent' seen in the wild)
+    version_number = data[ptr]
+    ptr += 1
+    if version_number != 4:
+        raise Exception("life is bad and I don't support this signature packet")
+
+    # here, we want to make sure the signature type is indeed PKCSV1.5 with RSA
+    signature_type = data[ptr]
+    ptr += 1
+    if signature_type != 0:
+        raise Exception("life is bad and I don't support this signature type")
+
+    signature_algorithm = data[ptr]
+    ptr += 1
+
+    hash_algorithm = data[ptr]
+    ptr += 1
+
+    if signature_algorithm != 1 or hash_algorithm != 8:
+        raise Exception("This library only supports rsa+pkcsv1.5 and sha256");
+
+    # obtain the hased octets.
+    hashed_octet_count = struct.unpack(">H", data[ptr:ptr+2])[0]
+    ptr += 2
+    hashed_subpackets = data[ptr:ptr+hashed_octet_count]
+    # check wether we were actually able to read this much hashed octets
+    if len(hashed_subpackets) != hashed_octet_count:
+        raise Exception("this signature packet is missing hashed octets!")
+    ptr += hashed_octet_count
+    other_headers_ptr = ptr
+
+    # we don't need this, but we will still parse them for the sake of 
+    # getting a keyid, this should be smarter down the line
+    # FIXME
+    unhashed_octet_count = struct.unpack(">H", data[ptr: ptr + 2])[0]
+    ptr += 2
+
+    unhashed_subpackets = data[ptr:ptr+unhashed_octet_count]
+    ptr += unhashed_octet_count
+
+    left_hash_bits = struct.unpack(">H", data[ptr:ptr+2])[0]
+    print("left hash bits {}".format(hex(left_hash_bits)))
+    ptr += 2
+
+    # Notice the /8 at the end, this length is the bitlength, not the length of
+    # the data in bytes (as len reports it)
+    signature_length = _get_mpi_length(data[ptr:ptr+2])
+    ptr += 2
+    signature = data[ptr:ptr + signature_length]
+    if len(signature) != signature_length:
+        raise Exception("This signature was truncated!")
+
+    return {
+        'keyid': "0x{}".format(binascii.hexlify(unhashed_subpackets).decode('ascii')),
+        'other-headers': binascii.hexlify(data[3:other_headers_ptr]).decode('ascii'),
+        'signature': binascii.hexlify(signature).decode('ascii')
+    }
+
+def _get_mpi_length(data):
+    bitlength = int(struct.unpack(">H", data)[0])
+
+    # ghetto ceil because ceil is the devil
+    return int((bitlength - 1)/8) + 1
+
+if __name__ == "__main__":
+    data = b'ayylmao\n'
+    signature = gpg_sign_object(data)
+    print(signature)
+
+    pubkey = gpg_export_pubkey("903BAB73640EB6D65533EFF3468F122CE8162295")
```

Comment:
this is my keyid, we still need to be clever-er at this, but this test code would work on your machine if you add the keyid of your default key in this line.